components/datadog/agent/ecsFargate.go

```diff
@@ -91,15 +87,11 @@ func ECSFargateLinuxContainerDefinition(e config.Env, image string, apiKeySSMPar
 // ECSFargateWindowsContainerDefinition returns the container definition for the Windows agent running on ECS Fargate.
 // Firelens is not supported. Logs could be collected if sent to cloudwatch using the `awslogs` driver. See:
 // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/tutorial-deploy-fluentbit-on-windows.html
-func ECSFargateWindowsContainerDefinition(e config.Env, image string, apiKeySSMParamName pulumi.StringInput, fakeintake *fakeintake.Fakeintake) *ecs.TaskDefinitionContainerDefinitionArgs {
-	if image == "" {
-		image = dockerAgentFullImagePath(e, "public.ecr.aws/datadog/agent", "latest", false)
-	}
-
+func ECSFargateWindowsContainerDefinition(e config.Env, apiKeySSMParamName pulumi.StringInput, fakeintake *fakeintake.Fakeintake) *ecs.TaskDefinitionContainerDefinitionArgs {
 	return &ecs.TaskDefinitionContainerDefinitionArgs{
 		Cpu:       pulumi.IntPtr(0),
 		Name:      pulumi.String("datadog-agent"),
-		Image:     pulumi.String(image),
+		Image:     pulumi.String(dockerAgentFullImagePath(e, "public.ecr.aws/datadog/agent", "latest", false)),
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
See my previous comment for the context of this suggestion:
```suggestion
		Image:     pulumi.String(dockerAgentFullImagePath(e, "", "", false)),
```